drivers/sensor/adi/adxl362/adxl362_decoder.c

```diff
@@ -20,11 +24,20 @@ static const uint32_t accel_period_ns[] = {
 	[ADXL362_ODR_400_HZ] = UINT32_C(1000000000) / 400,
 };
 
-static const int32_t range_to_scale[] = {
-	/* See table 1 in specifications section of datasheet */
-	[ADXL362_RANGE_2G] = ADXL362_ACCEL_2G_LSB_PER_G,
-	[ADXL362_RANGE_4G] = ADXL362_ACCEL_4G_LSB_PER_G,
-	[ADXL362_RANGE_8G] = ADXL362_ACCEL_8G_LSB_PER_G,
+static const uint32_t range_to_shift[] = {
+	[ADXL362_RANGE_2G] = 5,
+	[ADXL362_RANGE_4G] = 6,
+	[ADXL362_RANGE_8G] = 7,
+};
+
+/* (1 / sensitivity) * (pow(2,31) / pow(2,shift)) * (unit_scaler) */
+static const uint32_t qscale_factor[] = {
+	/* (1.0 / ADXL362_ACCEL_2G_LSB_PER_G) * (2^31 / 2^5) * SENSOR_G / 1000000 */
```

Comment:
Where is the division by 1000000 coming in? The fixed point value can already represent fractions to account for things so this division shouldn't be needed unless I've missed something

Reply:
Division is because SENSOR_G macro is 9806650LL and it is in micro m/s^2 units and we need m/s^2.